recipes/emsdk_installer/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+    description = "Emscripten is an Open Source LLVM to JavaScript compiler"
+    url = "https://github.com/bincrafters/conan-emsdk_installer"
+    homepage = "https://github.com/kripken/emscripten"
+    topics = ("conan", "emsdk", "emscripten", "installer", "sdk")
+    license = "MIT"
+
+    short_paths = True
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "emsdk-%s" % self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _run(self, command):
+        self.output.info(command)
+        self.run(command)
+
+    @staticmethod
+    def _create_dummy_file(directory):
+        if not os.path.isdir(directory):
+            os.makedirs(directory)
+        with open(os.path.join(directory, "dummy"), "w") as f:
+            f.write("\n")
+
+    @staticmethod
+    def _touch(filename):
+        if not os.path.isfile(filename):
+            with open(filename, "w") as f:
+                f.write("\n")
+
+    @staticmethod
+    def _chmod_plus_x(filename):
+        if os.name == 'posix':
+            os.chmod(filename, os.stat(filename).st_mode | 0o111)
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            emsdk = 'emsdk.bat' if os.name == 'nt' else './emsdk'
+            if os.path.isfile("python_selector"):
+                self._chmod_plus_x("python_selector")
+            self._chmod_plus_x('emsdk')
+            self._run('%s update' % emsdk)
+            if os.path.isfile("python_selector"):
+                self._chmod_plus_x("python_selector")
+            self._chmod_plus_x('emsdk')
+
+            self._run('%s install %s' % (emsdk, self.version))
+            self._run('%s activate %s --embedded' % (emsdk, self.version))
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        self.copy(pattern='*', dst='.', src=self._source_subfolder)
+        emsdk = self.package_folder
+        emscripten = os.path.join(emsdk, 'upstream', 'emscripten')
+        toolchain = os.path.join(
+            emscripten, 'cmake', 'Modules', 'Platform', 'Emscripten.cmake')
+        # allow to find conan libraries
+        tools.replace_in_file(toolchain,
+                              "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)",
+                              "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)")
+        tools.replace_in_file(toolchain,
+                              "set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)",
+                              "set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)")
+        tools.replace_in_file(toolchain,
+                              "set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)",
+                              "set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE BOTH)")
+
+    def _define_tool_var(self, name, value):
+        suffix = '.bat' if os.name == 'nt' else ''
```

Comment:
Don't use `os`. `settings.os` determines the os on which `emscripten` is supposed to be built for.
```suggestion
        suffix = '.bat' if self.settings.os == "Windows"
```